src/App.tsx

```diff
@@ -1,26 +1,68 @@
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 import { Loader, TodoFilter, TodoList, TodoModal } from './components';
+import { useEffect, useState, useMemo } from 'react';
+import { useDispatch } from 'react-redux';
+import { AppDispatch } from './app/store';
+import { todosSlice } from './features/todos';
+import { getTodos } from './api';
+import { useAppSelector } from './app/hook';
+import { Todo } from './types/Todo';
 
-export const App = () => (
-  <>
-    <div className="section">
-      <div className="container">
-        <div className="box">
-          <h1 className="title">Todos:</h1>
+export const App = () => {
+  const [isLoading, setIsLoading] = useState(false);
+  const dispatch = useDispatch<AppDispatch>();
+  const todos = useAppSelector(state => state.todosReducer);
+  const currentTodo = useAppSelector(state => state.currentTodoReducer);
+  const filterBy = useAppSelector(state => state.filterReducer.status);
+  const query = useAppSelector(state => state.filterReducer.query);
 
-          <div className="block">
-            <TodoFilter />
-          </div>
+  useEffect(() => {
+    setIsLoading(true);
+    getTodos()
+      .then(todos => dispatch(todosSlice.actions.setTodos(todos)))
+      .finally(() => setIsLoading(false));
+  }, [dispatch]);
+
+  const preparedTodos = useMemo(() => {
+    return todos
+      .filter((todo: Todo) => {
+        switch (filterBy) {
+          case 'active':
+            return !todo.completed;
+
+          case 'completed':
+            return todo.completed;
+
+          default:
+            return todo;
+        }
+      })
+      .filter(todo =>
+        todo.title.toLowerCase().trim().includes(query.trim().toLowerCase()),
+      );
+  }, [filterBy, todos, query]);
+
+
+  return (
+    <>
+      <div className="section">
+        <div className="container">
+          <div className="box">
+            <h1 className="title">Todos:</h1>
+
+            <div className="block">
+              <TodoFilter />
+            </div>
 
-          <div className="block">
-            <Loader />
-            <TodoList />
+            <div className="block">
+              {isLoading ? <Loader /> : <TodoList todos={preparedTodos} />}
```

Comment:
The conditional rendering of `<Loader />` and `<TodoList />` is correct, but ensure that the `Loader` component is properly implemented to handle the loading state.